Decentralized-Autonomous-Organizations.md

```diff
@@ -819,6 +822,37 @@ export default function Home() {
       window.alert(error.data.message);
     }
   };
+  
+  // Helper function to fetch owner of the DAO Contract
+  const getOwner = async () => {
+    try {
+      const provider = await getProviderOrSigner();
+      const daoContract = getDaoContractInstance(provider);
+      const _owner = await daoContract.owner();
+      const signer = await getProviderOrSigner(true);
+      const address = await signer.getAddress();
+      if (address.toLowerCase() === _owner.toLowerCase()) {
+        setIsOwner(true);
+      }
+    } catch (err) {
+      console.error(err);
+    }
+  };
+  
+  // Calls the `withdrawEther` function in the contract
+  const withdrawEther = async () => {
+    try {
+      const signer = await getProviderOrSigner(true);
+      const daoContract = getDaoContractInstance(signer);
+      const tx = await daoContract.withdrawEther();
+      setLoading(true);
+      await tx.wait();
+      setLoading(false);
+      await getOwner();
```

Comment:
I don't see a reason why you'd want to run `getOwner()` again here. Instead, maybe you'd want to run the `getDAOTreasuryBalance()` to update the Balance in the contract